src/main/java/io/cdap/plugin/aws/sqs/source/SqsSourceUtil.java

```diff
@@ -0,0 +1,210 @@
+ * the License.
+ */
+
+package io.cdap.plugin.aws.sqs.source;
+
+import com.amazonaws.services.sqs.AmazonSQS;
+import com.amazonaws.services.sqs.model.Message;
+import com.amazonaws.services.sqs.model.ReceiveMessageRequest;
+import com.amazonaws.services.sqs.model.SetQueueAttributesRequest;
+import io.cdap.cdap.api.data.format.FormatSpecification;
+import io.cdap.cdap.api.data.format.RecordFormat;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.etl.api.streaming.StreamingContext;
+import io.cdap.cdap.format.RecordFormats;
+import io.cdap.plugin.aws.sqs.exception.SqsInitializationException;
+import io.cdap.plugin.aws.sqs.util.SqsUtil;
+import org.apache.spark.storage.StorageLevel;
+import org.apache.spark.streaming.api.java.JavaDStream;
+import org.apache.spark.streaming.api.java.JavaReceiverInputDStream;
+import org.apache.spark.streaming.receiver.Receiver;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Util method for {@link SqsSource}.
+ *
+ * <p>This class contains methods for {@link SqsSource} that require spark classes because
+ * during validation spark classes are not available. Refer CDAP-15912 for more information.
+ */
+final class SqsSourceUtil {
+  private static final Logger LOG = LoggerFactory.getLogger(SqsSourceUtil.class);
+
+  private SqsSourceUtil() {
+    // no-op
+  }
+
+  static JavaDStream<StructuredRecord> getJavaDStream(StreamingContext context, SqsSourceConfig conf) {
+    return context.getSparkStreamingContext().receiverStream(SqsSourceUtil.getReceiver(conf));
+  }
+
+  /**
+   * Gets {@link JavaReceiverInputDStream} for {@link SqsSource}.
+   *
+   * @param conf {@link SqsSourceConfig} config
+   */
+  private static Receiver<StructuredRecord> getReceiver(SqsSourceConfig conf) {
+    return new Receiver<StructuredRecord>(StorageLevel.MEMORY_ONLY()) {
+      @Override
+      public StorageLevel storageLevel() {
+        return StorageLevel.MEMORY_ONLY();
+      }
+
+      @Override
+      public void onStart() {
+        new Thread() {
+          @Override
+          public void run() {
+            AmazonSQS sqs = null;
+            try {
+              sqs = SqsUtil.getSqsClient(conf.getAuthenticationMethod(), conf.getAccessId(),
+                conf.getAccessKey(), conf.getSqsEndpoint(), conf.getRegion().getName(), conf.getQueueName());
+            } catch (SqsInitializationException e) {
+              throw new RuntimeException(e);
+            }
+
+            while (!isStopped()) {
+              try {
+                List<Message> messages = fetchMessages(sqs, conf);
+                LOG.debug("messages.size() = {}", messages.size());
+                for (Message message : messages) {
+                  store(toRecord(message, conf));
```

Comment:
How does `store()` handle null values? This logic implies it just ignores them, is that correct?

Reply:
As you pointed out it is handling null values by ignoring and writing an error log with details. Can you please how do we handle this case more effectively?

Reply:
If it ignores the null values then that should be fine. I just wanted to make sure those werent being stored.